core/shared/src/main/scala/monocle/Lens.scala

```diff
@@ -251,6 +251,10 @@ object Lens {
   def codiagonal[S]: Lens[S \/ S, S] =
     PLens.codiagonal
 
+  /** creates a [[Lens]] from a Store comonad coalgebra */
+  def storeFCoalg[F[_], S, A](f: S => F[S])(implicit CS: ComonadStore[F, A]): Lens[S, A] =
```

Comment:
It's a *coalgebra* that commutes with the *comonad* operations. The idea can be found [here](https://www.schoolofhaskell.com/user/tel/lenses-from-scratch) and especially [here](https://bartoszmilewski.com/2015/07/13/from-lenses-to-yoneda-embedding/) (*The Algebraic Picture* section). It seems that the *comonad coalgebra* laws have a direct correspondence with the *lens* ones. 

As I said in the description, I've been toying around with the counterpart constructor for *Traversals*, but still don't know if traversal laws hold so nicely in that context (comonad coalgebras where `FreeAp[Store[A, ?], ?]` is the functor).